erdjs/src/dapp/walletProvider.ts

```diff
@@ -0,0 +1,237 @@
+        return this.waitForRemote();
+    }
+
+    /**
+     * Returns if the wallet iframe is up and running
+     */
+    isInitialized(): boolean {
+        if (!this.mainFrame) {
+            return false;
+        }
+
+        return this.mainFrame.dataset.initialized === "true";
+    }
+
+    /**
+     * Unlike isInitialized, isConnected returns true if the user alredy went through the login process
+     *  and has the wallet session active
+     */
+    async isConnected(): Promise<boolean> {
+        if (!this.mainFrame) {
+            return false;
+        }
+
+        const {contentWindow} = this.mainFrame;
+        if (!contentWindow) {
+            return false;
+        }
+
+        return new Promise((resolve, reject) => {
+            contentWindow.postMessage({
+                type: DAPP_MESSAGE_IS_CONNECTED,
+            }, this.walletUrl);
+
+            const timeout = setTimeout(_ => reject('window not responding'), 5000);
+
+            const isConnected = (ev: IDappMessageEvent) => {
+                if (!this.isValidWalletSource(ev.origin)) {
+                    return;
+                }
+
+                const {data} = ev;
+                if (data.type !== DAPP_MESSAGE_IS_CONNECTED) {
+                    return;
+                }
+
+                clearTimeout(timeout);
+                window.removeEventListener('message', isConnected.bind(this));
+                return resolve(data.data);
+            };
+
+            window.addEventListener('message', isConnected);
+        });
+    }
+
+    /**
+     * Fetches the login hook url and redirects the client to the wallet login.
+     */
+    async login(): Promise<string> {
+        if (!this.mainFrame) {
+            return '';
+        }
+
+        const {contentWindow} = this.mainFrame;
+        if (!contentWindow) {
+            return '';
+        }
+
+        return new Promise<string>((resolve, reject) => {
+            contentWindow.postMessage({
+                type: DAPP_MESSAGE_CONNECT_URL,
+            }, this.walletUrl);
+
+            const timeout = setTimeout(_ => reject('connect url not responding'), 5000);
+            const connectUrl = (ev: IDappMessageEvent) => {
+                if (!this.isValidWalletSource(ev.origin)) {
+                    return;
+                }
+
+                const {data} = ev;
+                if (data.type !== DAPP_MESSAGE_CONNECT_URL) {
+                    return;
+                }
+
+                clearTimeout(timeout);
+                window.removeEventListener('message', connectUrl.bind(this));
+                return resolve(data.data.toString());
+            };
+
+            window.addEventListener('message', connectUrl);
+        }).then((connectionUrl: string) => {
+            const redirectParts = connectionUrl.split('=');
+            const redirectAddress = redirectParts[redirectParts.length - 1];
+            if (redirectParts.length < 2 || !redirectAddress.startsWith('erd1')) {
+
```

Comment:
Do we want something special in this case?

Reply:
No, removed the code